kernel/timeout.c

```diff
@@ -169,29 +162,46 @@ void z_clock_announce(s32_t ticks)
 #endif
 
 	k_spinlock_key_t key = k_spin_lock(&timeout_lock);
+	struct _timeout *t = first();
 
-	announce_remaining = ticks;
+	curr_tick += ticks;
 
-	while (first() != NULL && first()->dticks <= announce_remaining) {
-		struct _timeout *t = first();
-		int dt = t->dticks;
+	if (t) {
```

Comment:
Could we replace this long `if` by early exit `if (!t)`?

Reply:
Not cleanly, since there's work to do after the conditional in both cases.  It could be switched to a loop, though, if we want to support immediate callbacks for newly scheduled timeouts that are known to be late.

Reply:
I suppose, though, we could do it as a `goto out`.  Lemme see how that looks.